tests/test_filters.py

```diff
@@ -86,6 +88,38 @@ def test_longdate_datetime_with_tz(self):
         with self.app.test_request_context('/'):
             assert filters.longdate(testdate) == testdate.strftime('%e %B %Y')
 
+    def test_date_filter(self):
+        with self.app.test_request_context('/'):
+            assert filters.date_filter(self.date, 'yyyy-MM-dd', usertz=False) == self.date.strftime('%Y-%m-%d')
+
+    def test_date_localized_short_hi(self):
+        with self.app.test_request_context('/', headers={'Accept-Language': 'hi'}):
+            assert filters.date_filter(self.date, format='short') == u'1/1/20'
+
+    def test_date_localized_medium_hi(self):
+        with self.app.test_request_context('/', headers={'Accept-Language': 'hi'}):
+            assert filters.date_filter(self.date, format='medium') == u'1 जन॰ 2020'
+
+    def test_date_localized_long_hi(self):
+        with self.app.test_request_context('/', headers={'Accept-Language': 'hi'}):
+            assert filters.date_filter(self.date, format='long') == u'1 जनवरी 2020'
+
+    def test_time_localized_hi(self):
+        with self.app.test_request_context('/', headers={'Accept-Language': 'hi'}):
+            assert filters.datetime_filter(self.datetime, format='medium') == u'1 जन॰ 2020, 12:00:00 am'
+
+    def test_month_localized_hi(self):
+        with self.app.test_request_context('/', headers={'Accept-Language': 'hi'}):
+            assert filters.date_filter(self.date, "MMMM") == u'जनवरी'
+
+    def test_month_localized_en(self):
+        with self.app.test_request_context('/'):
+            assert filters.date_filter(self.date, "MMMM") == 'January'
+
+    def test_time_localized(self):
+        with self.app.test_request_context('/', headers={'Accept-Language': 'hi'}):
+            assert filters.time_filter(self.datetime, format='medium') == '12:00:00 am'
+
```

Comment:
Missing tests:

1. Passing in a `locale` parameter that differs from `Accept-Language` header.
2. Timezone rendering with or without `usertz` parameter.
3. Tests with a `time` object.
4. Datetimes with a timezone that is (a) in UTC, and (b) elsewhere.